src/components/CommentList.jsx

```diff
@@ -13,7 +13,7 @@ const CommentList = () => {
       setComments(comments);
       setIsLoading(false);
     });
-  }, []);
+  }, [comments]);
```

Comment:
How come you've added comments to this useEffect dependency array?
Remember that when something in the dependency array chances, it triggers another request. Looks like this is what is causing your repeating requests